### PR TITLE
cache: dump cache on runner exception

### PR DESCRIPTION
**Pull request by ppanero**
- closes https://github.com/inveniosoftware/invenio-rdm-migrator/issues/61

### CODE REVIEW

**Review by alejandromumo — APPROVED** [review]

File: invenio_rdm_migrator/streams/runner.py [hunk 1]
```diff
@@ -23,39 +26,46 @@ def _read_config(self, filepath):
...
+                            ),
+                        )
                     )
-                )
+        finally:
+            for name, cache in self.cache.items():
+                if name == "communities":  # FIXME: implement communities cache
+                    continue
+                cache_file = self.cache_dir / f"{name}.json"
+                cache.dump(cache_file)
```
Comment (slint) [inline invenio_rdm_migrator/streams/runner.py:68]: Shouldn't this be done inside `.run()` below? I think here we only initialize the streams, but don't run anything.

**Review by zzacharo — APPROVED** [review]

File: invenio_rdm_migrator/streams/cache.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def __init__(self, filepath=None, validate=False):
         """Constructor."""
         self._data = {}
 
-        if filepath:  # load cache from file
+        if filepath and filepath.exists():  # load cache from file
```
Comment (slint) [inline invenio_rdm_migrator/streams/cache.py:21]: To reinforce what @alejandromumo mentioned, could we put some `print(f"loading cache from {fliepath}")` statements to know what's happening?

**Review by slint — COMMENTED** [review]
Nice! One "major" comment about when we dump the cache, and some print/logging

File: invenio_rdm_migrator/streams/runner.py (hunk 1 above)
Comment (ppanero) [reply invenio_rdm_migrator/streams/runner.py:68]: Ufff, yes, totally true. My bad.

File: invenio_rdm_migrator/streams/cache.py (hunk 2 above)
Comment (ppanero) [reply invenio_rdm_migrator/streams/cache.py:24]: done, added also for dumping:

e.g.
```
Loading cache from /Users/ppanero/Workspace/zenodo/demo/cache/parents.json 2023-05-05T11:27:50.212081
Finished loading cache 2023-05-05T11:27:50.212170
Cache loading took 0.00010991096496582031 seconds.
Loading cache from /Users/ppanero/Workspace/zenodo/demo/cache/records.json 2023-05-05T11:27:50.212217
Finished loading cache 2023-05-05T11:27:50.212278
Cache loading took 7.390975952148438e-05 seconds.
Stream started 2023-05-05T11:27:50.212548
Dumping cache to /Users/ppanero/Workspace/zenodo/demo/cache/parents.json 2023-05-05T11:27:50.212678
Finished dumping cache 2023-05-05T11:27:50.213631
Cache dumping took 0.0009357929229736328 seconds.
Dumping cache to /Users/ppanero/Workspace/zenodo/demo/cache/records.json 2023-05-05T11:27:50.213669
Finished dumping cache 2023-05-05T11:27:50.213820
Cache dumping took 0.0001399517059326172 seconds.
```

**Pull request merged**